packages/cli/test/unit/commands/promote.test.ts

```diff
@@ -0,0 +1,380 @@
+import chalk from 'chalk';
+import { client } from '../../mocks/client';
+import { defaultProject, useProject } from '../../mocks/project';
+import { Request, Response } from 'express';
+import promote from '../../../src/commands/promote';
+import { LastAliasRequest } from '@vercel-internals/types';
+import { setupUnitFixture } from '../../helpers/setup-unit-fixture';
+import { useDeployment } from '../../mocks/deployment';
+import { useTeams } from '../../mocks/team';
+import { useUser } from '../../mocks/user';
+import sleep from '../../../src/util/sleep';
+
+jest.setTimeout(60000);
+
+describe('promote', () => {
+  it('should error if cwd is invalid', async () => {
+    client.setArgv('promote', '--cwd', __filename);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      'Error: Support for single file deployments has been removed.'
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if timeout is invalid', async () => {
+    const { cwd } = initPromoteTest();
+    client.setArgv('promote', '--yes', '--cwd', cwd, '--timeout', 'foo');
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput('Error: Invalid timeout "foo"');
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if invalid deployment name', async () => {
+    const { cwd } = initPromoteTest();
+    client.setArgv('promote', '????', '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      'Error: The provided argument "????" is not a valid deployment'
+    );
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if deployment not found', async () => {
+    const { cwd } = initPromoteTest();
+    client.setArgv('promote', 'foo', '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput('Fetching deployment "foo" in ');
+    await expect(client.stderr).toOutput(
+      'Error: Error: Can\'t find the deployment "foo" under the context'
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should show status when not rolling back', async () => {
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** 

```suggestion
  it('should show status when not promoting', async () => {
```